README.txt

 Kana no quiz - README
 version 2.0 - in development 
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

 Introduction
 ============

 This is ``Kana no quiz", a little educational tool, simple yet efficient, to
memorize Japanese kana (hiragana & katakana) pronounciation in an easy way,
and thus accomplish the first step in learning the language, without much pain.

 The full list of kana is implemented, including additional katakana used in
modern language. Default used transcription system is the Hepburn's (most
common world-wide), but it does also support Kunrei-shiki, Nihon-shiki and
Polivanov (cyrillic) systems.

 Written in Python, this program is cross platform: the same package works on
Unix (GNU/Linux, FreeBSD, Mac OS X...) and Windows, using the same GTK+
interface on every platform.

 How does it work?
 -----------------

 The method is quite simple: a kana appear randomly and the user has to find
its transcription ; either by choosing the correct answer from a random list
containing several possibilities (number is selectable) or by directly typing
the answer through a text entry. This second method of course is more
complicated.

 Progressive learning
 --------------------
 
 Through its option panel and full kana tables, Kana no quiz allows you to tune
up the quiz precisely, like choosing which group of kana you want to train
yourself upon.

 First, hiragana and katakana are separated, both are then divided into big
sets of kana according to their type (``Basic", ``Modified", ``Combined"; and
a special ``Additional" set for katakana only) which are thus composed of
various portions mostly containing 5/6 kana, following the traditional alphabet
order (i.e.: ``ka ki ku ke ko").

 According to your need and level, you may select one or more portions of the
same or different sets, whole sets, all the kana... Once you feel comfortable
(you can check it with the results displayed at the end of each quiz), you may
also increase the number of possible answers in the random list, or even choose
to answer directly through the text entry. Difficulty can be adapted to your
skill: this is progressive learning.

 From the complete beginner who has to start easily and learn step by step, to
the wizard who just wants to confirm his/her absolute master of the kana,
everyone should get satisfated with Kana no quiz!


 Use Kana-no-quiz
 ================

 Prerequisites
 -------------

 To run fine, Kana no quiz requires the following softwares to be priorly
installed on your computer:

* Python - `http://www.python.org/'.
* For the GTK+ interface:
  * The Gimp Tool Kit - `http://www.gtk.org/'.
  * PyGTK - `http://www.pygtk.org/'.

Optional:
  * Psyco (to increase program speed performance) - `http://psyco.sf.net/'
  * SDL & PyGame (to hear kana prononciation sounds) - `http://www.libsdl.org/'
    and `http//www.pygame.org/'.

(Windows user, please note that *all* those dependencies are included in the
Kana no quiz all-in-one installer)

 Program installation
 --------------------
 
 This process depends on your package type and platform.

 On Windows, simply install Kana-no-quiz using one of our graphical, binary
installers (either the single Python package or the popular ``all-in-one").

 On GNU/Linux, like other Unixes, if not using any distro-specific package but
the classic source code, simply enter `python setup.py install' as root to
install Kana-no-quiz.

 Launching
 ---------

 On Windows, just click on the shortcut created in the Start Menu.

 On Unix systems, simply enter the `kana-no-quiz' command.

 Enjoy. ;)

 Some details
 ============

 I18n
 ----

 Thanks to our contributors, Kana no quiz is currently translated into:
English, French, German (by Florian Niemann), Portuguese of Brazil (by Matheus
Villela), Russian (by Aleksej R. Serdyukov), Serbian (by Dejan Danilović) and
Swedish (by Markus Fellnert).

 Feel free to help that list growing up!

 Kana image sources
 ------------------
 
 The kana images are bitmap exports of SVG files we modelized by ourselves,
using Inkscape, with the hope they could also be reused for other purposes,
by people who enjoy them. So, the source SVGs have been made publicly available
(see our website), under the conditions indicated above.


 Licensing
 =========

 Kana no quiz is free software with its source code released under the terms of
the GNU General Public License (see `GPL.txt').

 Meanwhile, the whole multimedia content (except the traditional black and
white kana image theme that belongs to KanaTest, released under the GPL) is,
for its part, released under the terms of the Free Art license (see `FAL.txt').


 Stay tuned
 ==========

 Kana no quiz is a project conducted by the Choplair-network, you can get
lastest information and update from our website:
	`http://www.choplair.org/'


 That's all for now, have a lot of fun. ;)
	Choplair



CHANGES.txt

 Kana no quiz - Changelog
~~~~~~~~~~~~~~~~~~~~~~~~~~

* Version 1.9.5 (2009-07-18):

- Added Spanish and Galician translations, contributed by
Rafael Lopez Garcia.
- Some little bug fixes.


* Version 1.9 (2007-09-13):

- Vocal pronouncing of all kana (w/ male & female voices).
- Switchable kana image theme (with the addition of a second
theme, taken from the GPL program "Kana Test").
- Faster loading and enhanced design of the big kana table
window.
- Updated Polivanov system (new transcripted kana).


* Version 1.7 (2006-10-15):

- Brand new & great kana table window providing a very
conveniant way to either view the full kana list (with trans-
cription, and respecting traditionnal order), select the portion
to be trained upon, or see high size kana image (by clicking
on thumbnails).
- New kana image scaling feature (user may choose the relative
size of the kana images which appear during the quiz).
- Optional use of Psyco to increase Kana no quiz's overall
speed. ;-)
- All images have now switched to PNG format instread of GIF.
- Removing the Tkinter GUI to keep back the GTK as the unique
interface for Kana no quiz.
- Various GUI design enhancements (including some little
choplair arworks realised by Fayane).
- Important code style cleanups plus programming design
optimizations (thanks to Jeremy Bowers' patches in particular).


* Version 1.5 (2006-05-12):

- System-wide installation & per-user configuration.
- More complete quiz result informations (wrong answer recap).
- Windows binary installation package.
- No more learning flow disturbance within a quiz using the
text-entry answering method.
- Added possibility to extend random answers selection range
from the same portion to the same kana set or kind.
- Added Polivanov's kana cyrillization system.
- New Russian translation (made by Aleksej R. Serdyukov).
- New German translation (made by Florian Niemann).
- A quit button added to the main window.


* Version 1.4b (2006-01-01):

- Language confusion between Serbian & Swedish with the
Tkinter interface corrected.
- Update of the Swedish translation.


* Version 1.4 (2005-12-14):

- You can now choose several portions of the same kana set to
train upon in the same time (if no portion selected in a set,
that one becomes disabled :p).
- Random list answer mode now permits up to five choices. :p
- Support for the 2 other popular romanization systems: the
Kunrei-shiki's and the Nihon-shiki's. (Because they are really
less used worldwide than the Hepburn's, choosing one of these
systems remains for people who know what they are doing.)
- Some informations are now displayed during the quiz (romani-
zation system used and question number) and you can choose to
stop it after every answer (useful for example if you realized
the quiz you began to perform is truly too difficult).
- Bug fix: no more endless quiz!
- Some improvements over the GTK+ interface, and replacement
of the depracted widgets.
- Addition of a marvellous icon (GTK+ version)!
- Translations update (except for Brazilian).


* Version 1.2d (2005-03-17):

- Addition of a Serbian translation.
- Some (English) spelling corrections.


* Version 1.2c (2005-02-14):

- A bug created in the last version has been fixed...
Now the ``kana engine" should be clean ! :-D


* Version 1.2b (2005-02-07):

- Bug fix : prevention, while doing the quiz using the random
list mode, of getting the right answer appearing twice in
the list.
- The interface type used by Kana no quiz is now indicated
in the ``About" window (in case that you don't see! ^^).


* Version 1.2 (2005-01-23):

- The Sweedish translation has been finaly updated.
- In the GTK version, we can choose the length of the quiz
from 1 to 200 questions.
- Addition of a new option to not repeat the same kana
during the quiz (as much as possible: a kana might appear
several time if the quiz lenght is bigger than the number of
selected kana).


* Version 1.1 (2004-09-25):

- Doing the quiz, the same kana now won't appear twice in
a row any more. :p
- A new ``Introduction" window which briefly introduce the
Kana writing system and what this program is aimed to.
- You can now choose to get trained only upon a portion of
kana rather than the full kana set, it may be helpful for
the beginers! ^_^;


* Version 1.0 (2004-07-29):

- Addition of the ``additional" katakana question set which
contains syllabes used for foreign words.
The sets are now: ``Basic" (formerly ``Single"), ``Modified",
``Contracted" (formerly ``Combined") & ``Additional" (for
katakana only). In other words: *ALL* kana have been (normally ;)
implemented into Kana no quiz. ^_^
- The answers can now be typed directly through a text entry
rather than selected from a list. It's harder! ^^;
- The option panel has been slighty redesigned.
- Program translation into Portuguese of Brazil.
Unfortunately, the Sweedish translation has not been updated. ;( 


* Version 0.9 (2004-05-31):

- New combined hiragana & katakana added.
- Questions are now divided in several sets (single/modified
/combined).
- At least one question set must be selected or a dialog will
warn you. :p
- Update of translations.


* Version 0.8 (2004-05-08):

- Forty-one kana added (total: 142). All the basic
syllables of both the katakana & hiragana syllabaries
are now implemented. :)


* Version 0.7 (2004-04-17):

- Forty-five new kana added (total: 101).
- Many previous kana have been enhanced!


* Version 0.6 (2004-04-03):

- Huge improvements of Tkinter interface (Windows)!
- Thirteen new kana added (total: 66).
- Better handling of the configuration file.
- Program language can be now changed on the fly from the
configuration panel.
- Bug fix: first param appeared as a comment in the
configuration file.
- Bug fix: results were shown just after the last question
without told whether the answer was right or wrong!


* Version 0.5 (2004-03-22):

- Internationalization (translations into French and Swedish).
- Seventeen new kana added (total: 53).

data/py/i18n.py
# -*- coding: utf-8 -*-
"""Kana no quiz!

   Copyleft 2003, 2004, 2005, 2006, 2007, 2008, 2009 Choplair-network.
   $Id$

   This program is free software; you can redistribute it and/or
   modify it under the terms of the GNU General Public License
   as published by the Free Software Foundation; either version 2
   of the License, or (at your option) any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License
   along with this program; if not, write to the Free Software
   Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.

"""
import gettext

class I18n:
   def __init__(self, path):
      self.currlang = str()
      self.localepath = path

   def setlang(self, lang):
      if lang == "en":
         global _
         def _(msg): return msg
      else: 
         try:   #Removing the global _ if the current language is english.
            if self.currlang == "en": del _
         except: pass

         gettext.translation("kana-no-quiz", self.localepath, [lang]).install()

      self.currlang = lang #Update the current language variable.

   def msg(self,num):
      # Tuple of commom strings.
      strings = (
         _("Kana no quiz!"),  # 0
         _("Introduction"),
         _("Options"),
         _("Start the quiz"),
         _("About"),
         _("Two different writing systems are commonly used by the Japanese: "
            "kanji & kana."),  # 5
         _("Kanji are complex ideograms retranscribing words whereas kana "
            "are simple symbols used as syllables which have to be "
            "combined to make words. Any word in Japanese can be written "
            "using kana, it is like an alphabet."),
         _("There are two kinds of kana: hiragana & katakana. Hiragana is "
            "the most-used traditional syllabary used to write Japanese "
            "words. Katakana is a larger syllabary, with more sounds, "
            "mostly used to write proper nouns and loanwords of foreign "
            "origin."),
         _("Kana no quiz helps you to memorize kana and their associated "
            "sound, using transcription systems. For more simplicity, "
            "syllables have been split in multiple sets, and you can even "
            "choose to learn only a portion of kana at a time."),
         _("Have a good practice ! ^_^"),
         _("OK"),  # 10
         _("What is this katakana?"),
         _("What is this hiragana?"),
         _("Right! ^^"),
         _("Wrong!"),
         _("The answer was %s."),  # 15
         _("Quiz results"),
         _("Question total: %i."),
         _("Good answers: %i."),
         _("Success rate: %i%%."),
         _("Here you can modify some quiz options."),  # 20
         _("Hiragana question sets"),
         _("Katakana question sets"),
         _("Basic"),
         _("Modified"),
         _("Contracted"),  # 25
         _("Additional"),
         _("Portions"),
         _("Hiragana"),
         _("Katakana"),
         _("Basic set"),  # 30
         _("Modified set"),
         _("Contracted set"),
         _("Additional set"),
         _("Additional katakana"),
         _("If you are novice, it may be interesting at the beginning to "
            "train yourself only on portions instead of the full kana set."),  # 35
         _("Select the kana portions you want to train yourself upon:"),
         _("Select all"),
         _("Answering method:"),
         _("A random list"),
         _("A text entry"),  # 40
         _("Random answer list size:"),
         _("%s choices"),
         _("Quiz length:"),
         _("Don't repeat the same kana\n(as much as possible)."),
         _("Language:"),  # 45
         _("English"),
         _("French"),
         _("Portuguese (Brazil)"),
         _("Serbian"),
         _("Swedish"),  # 50
         _("Save"),
         _("Cancel"),
         _("A Japanese kana memorization tool."),
         _("Version %s."),
         _("Kana no quiz is free software with its source code released under "
         "the terms of the GNU General Public License (see `GPL.txt').\nThe "
         "whole multimedia content is released under the terms of the Free "
         "Art license (see `FAL.txt') with the exception of the kana image "
            "theme that was adapted from \"Kana Test\", released under the "
            "GNU GPL as well."),  # 55
         _("Credits"),
         _("Main development: Choplair.\n"
            "Code contribution: Jeremy Bowers."),
         _("Close"),
         _("Warning"),
         _("Please select at least one kana portion to start the quiz."),  # 60
         _("Transcription system:"),
         _("Hepburn (most used)"),
         _("Kunrei-shiki"),
         _("Nihon-shiki"),
         _("Information"),  # 65
         _("Programming"),
         _("Question {0} of {1}."),
         _("%s system."),
         _("Stop"),
         _("German"),  # 70
         _("Quit"),
         _("Unrecognized katakana: %s."),
         _("Unrecognized hiragana: %s."),
         _("Russian"),
         _("Random answer selection range:"),  # 75
         _("Same kana portions"),
         _("Same kana set"),
         _("Same kana kind"),
         _("Polivanov (cyrillic)"),
         _(" questions."),  # 80
         _("Translations"),
         _("Brazilian: Matheus Villela.\n"
            "French: Choplair.\n"
            "Galician: Rafael López García.\n"
            "German: Florian Niemann.\n"
            "Russian: Aleksej R. Serdyukov.\n"
            "Serbian: Dejan Danilovic.\n"
            "Spanish: Rafael López García.\n"
            "Swedish: Markus Fellnert."),
         _("Kana"),
         _("Answering"),
         _("Sound"),  #85
         _("Misc."),
         _("Kana tables"),
         _("Welcome to Kana no quiz version %s."),
         _("Crappy!"),
         _("Average"),  # 90
         _("Good"),
         _("Excellent!"),
         _("Here comes the complete kana tables, sorted respecting the "
            "Japanese traditional order (except for additional katakana "
            "that haven't been standardized)."),
         _("You may enjoy full kana list overview (with transcription), "
            "select the portions you want to be trained upon, and see larger "
            "image when clicking on a thumbnail."),
         _("Kana image scale:"),  # 95
         _("Small"),
         _("Medium"),
         _("Large"),
         _("Kana image theme:"),
         _("Modern colorful"),  # 100
         _("Traditionnal black & white"),
         _("Kana vocal pronouncing"),
         _("Female: Myrtille Chouard.\n"
           "Male: Pachilor."),
         _("Kana image themes"),
         _("Modern colorful: Ms. Marie-Claire.\n"
           "Traditional black & white:\n"
           "adapted from \"Kana Test\"."),  # 105
         _("Artworks"),
         _("Fayanne & Choplair."),
         _("Kana vocal pronouncing:"),
         _("Female"),
         _("Male"),  # 110
         _("Both alternately"),
         _("Spanish"),
         _("Galician"),
         _("Please restart Kana no quiz for the language change to take "
            "effect."),
         )

      return strings[num]
